torchtext/csrc/regex.cpp

```diff
@@ -0,0 +1,48 @@
+#include <regex>
+#include <string>
+#include <torch/script.h>
+
+namespace torchtext {
+namespace {
+
+struct Regex : torch::CustomClassHolder {
+private:
+  std::regex re_;
+
+public:
+  // re_str_ holds the serialized regex string passed at the initialization.
+  // We need this because we need to be able to serialize the model so that we
+  // can save the scripted object. pickle will get the
+  // serialized model from this re_str_ member, thus it needs to be public.
+  std::string re_str_;
+
+  Regex(const std::string &re_str) { UpdateRe(re_str_); }
+
+  void UpdateRe(const std::string &re_str) {
```

Comment:
Discussed about this offline. Seems simpler to create one object and update the regex pattern instead of creating an arbitrary number of objects using a factory method.